master/buildbot/db/model.py

```diff
@@ -759,6 +759,17 @@ def thd(engine):
             return db_version == repo_version
         return self.db.pool.do_with_engine(thd)
 
+    def enableForeignKeys(self):
+        # foreign keys is a recent feature of sqlite3, it is still disabled by default
+        # https://www.sqlite.org/pragma.html#pragma_foreign_key_check
```

Comment:
I tried to run a typoed plasma and got no error. This is why I kept it like
this. Ill look at it more seriously

Le lun. 29 févr. 2016 14:59, Dustin J. Mitchell <notifications@github.com>
a écrit :

> In master/buildbot/db/model.py
> <https://github.com/buildbot/buildbot/pull/2024#discussion_r54412150>:
>
> > @@ -759,6 +759,17 @@ def thd(engine):
> >              return db_version == repo_version
> >          return self.db.pool.do_with_engine(thd)
> >
> > +    def enableForeignKeys(self):
> > +        # foreign keys is a recent feature of sqlite3, it is still disabled by default
> > +        # https://www.sqlite.org/pragma.html#pragma_foreign_key_check
>
> From the docs, it looks like yes. In that case, should this function catch
> the exception and log it, if the support is not present?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/2024/files#r54412150>.
>